dashboard/test/controllers/api_controller_test.rb

```diff
@@ -396,11 +396,14 @@ def get_student_response(script, level, lesson, student_number)
 
     user_level_data = student_5_response['user_level_data']
     user_level = UserLevel.find_by(user_level_data)
+    assert_equal false, user_level.submitted?
     assert_equal true, student_5_response['locked']
+    assert_equal true, user_level.locked
+
+    # we only show as readonly_answers if we're not also locked, so the
+    # response value is "false" even though the value in the db is "true"
+    assert_equal true, user_level.readonly_answers?
```

Comment:
as far as i can tell, this is the only test that was failing due to incorrect logic. you were asserting that both `student_5_response['readonly_answers']` _and_ `user_level.readonly_answers?` were false. but since `lesson.rb` takes the `show_as_locked` value into consideration and on the client `locked` and `readonly` can't both be true, we actually want to assert that the response value is `false` while the db value is still `true`. clarifying this was my primary motivation for adding the `show_as_readonly` helper.

(i reordered the assertions just because `locked` and `submitted?` don't need comment)

Reply:
Yes- I think this is part of what was so weird about this test case to begin with. Ending up with such a dissonant state (db doesn't match output) calls for a comment and an extra helper making it visible- so thanks for adding.